planner/core/planbuilder.go

```diff
@@ -1500,6 +1500,45 @@ func (b *PlanBuilder) buildDDL(node ast.DDLNode) (Plan, error) {
 			table:     v.Table.Name.L,
 			err:       nil,
 		})
+		if len(v.Specs) == 1 && v.Specs[0].Tp == ast.AlterTableRenameTable {
```

Comment:
I‘m not sure the `Specs[0]`, is it possible has `Specs[1]` or `Specs[...]` ?

Reply:
Of course, it could be. But now we only support just 1 specs, but I think you better need to iterate the specs array.

You can find the logic in the parser.y:

```
AlterTableStmt:
	"ALTER" IgnoreOptional "TABLE" TableName AlterTableSpecList
	{
		$$ = &ast.AlterTableStmt{
			Table: $4.(*ast.TableName),
			Specs: $5.([]*ast.AlterTableSpec),
		}
	}
```

It is a list.